syntax = "proto3";

enum HealthStatus {
    HEALTHY = 0;
    UNHEALTHY = 1;
}

message PingHealthESSIMToMSO {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is ESSIM unhealthy?
    repeated string activeSimulations = 3;
}

message PongHealthMSOToEssim {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is MSO unhealthy?
}

message PingHealthMSOToModel {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is MSO unhealthy?
}

message PongHealthModelToMSO {
    HealthStatus healthy = 1;
    repeated string reasons = 2; // Why is model unhealthy?
    int32 numberOfBids = 3;  // How many bids have been generated by model.
    int32 numberOfAllocations = 4;  // How many allocations have been allocated by model.

}
